src/com/projectkorra/projectkorra/firebending/Illumination.java

```diff
@@ -110,20 +142,47 @@ private void revert() {
 	}
 
 	private void set() {
-		final Block standingBlock = this.player.getLocation().getBlock();
-		final Block standBlock = standingBlock.getRelative(BlockFace.DOWN);
-		if (!isIgnitable(standingBlock)) {
-			return;
-		} else if (this.block != null && standingBlock.equals(this.block.getBlock())) {
-			return;
-		} else if (Tag.LEAVES.isTagged(standBlock.getType())) {
-			return;
-		} else if (standingBlock.getType().name().endsWith("_FENCE") || standingBlock.getType().name().endsWith("_FENCE_GATE") || standingBlock.getType().name().endsWith("_WALL") || standingBlock.getType() == Material.IRON_BARS || standingBlock.getType().name().endsWith("_PANE")) {
-			return;
+		if (MODERN) { //Light block implementation
+			Block eyeBlock = this.player.getEyeLocation().getBlock();
+			int level = lightLevel;
+			if (!eyeBlock.getType().isAir() && (this.block == null || !this.block.getBlock().equals(eyeBlock))) {
+				for (BlockFace face : new BlockFace[] {BlockFace.UP, BlockFace.DOWN, BlockFace.NORTH, BlockFace.SOUTH, BlockFace.EAST, BlockFace.WEST}) {
+					if (eyeBlock.getRelative(face).getType().isAir() || (this.block != null && this.block.getBlock().equals(eyeBlock.getRelative(face)))) {
+						eyeBlock = eyeBlock.getRelative(face);
+						level = lightLevel - 1; //Make the light level 1 less

```

Comment:
Why are you making the lightLevel less as opposed to resetting it to what it previously was?

Reply:
The aim here was to do the light with one level less due to it displaying one block from where is preferable

Reply:
I like dimming, brilliant.